examples/cluster/plot_cluster_comparison.py

```diff
@@ -35,17 +35,18 @@
 from sklearn.neighbors import kneighbors_graph
 from sklearn.preprocessing import StandardScaler
 
-np.random.seed(0)
-
 # ============
 # Generate datasets. We choose the size big enough to see the scalability
 # of the algorithms, but not too big to avoid too long running times
 # ============
 n_samples = 500
-noisy_circles = datasets.make_circles(n_samples=n_samples, factor=0.5, noise=0.05)
-noisy_moons = datasets.make_moons(n_samples=n_samples, noise=0.05)
+noisy_circles = datasets.make_circles(
+    n_samples=n_samples, factor=0.5, noise=0.05, random_state=8
+)
+noisy_moons = datasets.make_moons(n_samples=n_samples, noise=0.05, random_state=8)
 blobs = datasets.make_blobs(n_samples=n_samples, random_state=8)
-no_structure = np.random.rand(n_samples, 2), None
+rng = np.random.RandomState(42)
```

Comment:
I'd move this `rng` a few lines above and pass the same thing to `mane_moons` and `make_circles`

Reply:
How does this relate to the changing order and the RandomState object usage conversation we had at the workshop (cc @glemaitre as it was us talking)? I got the impression that whenever possible the `random_state` parameter is to be preferred? As far as I understand, it isn't directly possible to add it in the `no_structure` case, so we use `RandomState` instead of the `numpy` implementation. From a best practice perspective, is it different in this specific case? Or this is a style suggestion so to say? Thanks!

Reply:
Yes, I'm not saying not to use `random_state`, I'm suggesting:

```py
noisy_moons = datasets.make_moons(n_samples=n_samples, noise=0.05, random_state=rng)
```

Reply:
Right @adrinjalali sorry, I wasn't clear enough, I meant passing an integer to the `random_state ` parameter, instead of a `RandomState` object. From my conversation with @glemaitre at the event it felt like the first is preferred, we discussed exactly this. I am also happy to go with it as it's better to merge it than to stall it for a few weeks, just trying to understand.

Reply:
then I'll let @glemaitre comment as why :)

Reply:
@adrinjalali passing the `rng` instance, it will be mutated by each call. Potentially, if we change a line of code that alternates the `rng` in a future PR, we will change all subsequent behaviour of the function using `rng`. Passing an `int` will make sure that we don't change the behaviour in future PRs.

Basically, random-state instance dilemma stuff :)